rskj-core/src/main/java/org/ethereum/core/TransactionExecutor.java

```diff
@@ -526,13 +535,13 @@ public void finalization() {
 
         logger.trace("tx listener done");
 
-        if (vmTrace && program != null) {
+        if (this.programTraceProcessor.enabled() && program != null) {
```

Comment:
Could we either:
* hide the processor enabling/disabling inside the trace processor class, or
* remove the enabling/disabling from the trace processor class and handle it here

I think it's confusing to have to verify this condition in two places.

Reply:
The trace processor is injected "just-in-time" so now has the responsibility of inform the flag

Reply:
What do you mean by _just-in-time_? AFAIKS at the time the _programTraceProcessor_ instance is created, we already know if it should trace or not, and given that toggling it in runtime is not an expected feature I see this check here really confusing.